AdminWebsite/AdminWebsite.UnitTests/Helper/AdministratorRoleClaimsTest.cs

```diff
@@ -0,0 +1,117 @@
+            parsed.UserCaseTypes.Should().NotContainNulls();
+            parsed.UserCaseTypes.Should().HaveCountGreaterThan(0);
+            parsed.UserCaseTypes.Should().BeEquivalentTo(userRole.CaseTypes);
+        }
+
+        [Test]
+        public void should_return_empty_list_of_user_case_types()
+        {
+            var userRole = new UserRole
+            {
+                UserRoleType = UserRoleType.VhOfficer
+            };
+
+            var created = new AdministratorRoleClaims(userRole);
+            created.Should().NotBeNull();
+            created.UserCaseTypes.Should().NotContainNulls();
+            created.UserCaseTypes.Should().HaveCount(0);
+
+            var parsed = new AdministratorRoleClaims(created.Claims);
+            parsed.Should().NotBeNull();
+            parsed.UserCaseTypes.Should().NotBeNull();
+            parsed.UserCaseTypes.Should().NotContainNulls();
+            parsed.UserCaseTypes.Should().HaveCount(0);
+        }
+
+        [TestCase(UserRoleType.None, false)]
+        [TestCase(UserRoleType.VhOfficer, true)]
+        [TestCase(UserRoleType.CaseAdmin, false)]
+        [TestCase(UserRoleType.Individual, false)]
+        [TestCase(UserRoleType.Judge, false)]
+        [TestCase(UserRoleType.Representative, false)]
+        public void should_set_isvhofficeradministratorrole_property(UserRoleType role, bool expectedValue)
+        {
+            var userGroupData = new UserRole { UserRoleType = role };
+
+            var created = new AdministratorRoleClaims(userGroupData);
+            created.IsVhOfficerAdministratorRole.Should().Be(expectedValue);
+
+            var parsed = new AdministratorRoleClaims(created.Claims);
+            parsed.IsVhOfficerAdministratorRole.Should().Be(expectedValue);
+        }
+
+        [TestCase(UserRoleType.None, false)]
+        [TestCase(UserRoleType.VhOfficer, false)]
+        [TestCase(UserRoleType.CaseAdmin, true)]
+        [TestCase(UserRoleType.Individual, false)]
+        [TestCase(UserRoleType.Judge, false)]
+        [TestCase(UserRoleType.Representative, false)]
+        public void should_set_iscaseadministratorrole_property(UserRoleType role, bool expectedValue)
+        {
+            var userGroupData = new UserRole { UserRoleType = role };
+
+            var created = new AdministratorRoleClaims(userGroupData);
+            created.IsCaseAdministratorRole.Should().Be(expectedValue);
+
+            var parsed = new AdministratorRoleClaims(created.Claims);
+            parsed.IsCaseAdministratorRole.Should().Be(expectedValue);
+        }
+
+        [TestCase(UserRoleType.None, false)]
+        [TestCase(UserRoleType.VhOfficer, true)]
+        [TestCase(UserRoleType.CaseAdmin, true)]
+        [TestCase(UserRoleType.Individual, false)]
+        [TestCase(UserRoleType.Judge, false)]
+        [TestCase(UserRoleType.Representative, false)]
+        public void should_set_isadministratorrole_property(UserRoleType role, bool expectedValue)
```

Comment:
Mm, this makes a lot more sense to me (but then again, that might be because we've spent hours talking about it :P )